src/Symfony/Component/Routing/Loader/AnnotationClassLoader.php

```diff
@@ -205,6 +205,7 @@ protected function addRoute(RouteCollection $collection, $annot, array $globals,
             $this->configureRoute($route, $class, $method, $annot);
             if (0 !== $locale) {
                 $route->setDefault('_locale', $locale);
+                $route->setRequirement('_locale', $locale);
```

Comment:
Should be escaped with preg_quote()?

Reply:
Setting a default locale should not imply a requirement, this should be reverted here.

Reply:
IMO, a locale can't (and shouldn't) contain characters that need to be escaped.
WDYT ?

Reply:
That's a second reason why in should be escaped: filter your input or someone creative will find a way to hack your code.

Reply:
Agreed!